### DISCUSSION TITLE
Playwright Testing Vueform Builder

### DISCUSSION THREAD

**Discussion by victorvaldez5** [vueform/builder#36] [Questions] [answered]
Hello Everyone,

I'm facing an issue with license validation blocking our Playwright end-to-end tests. We are trying to create an end-to-end test for a page that uses the form builder, and when we go to access the Form Builder page, we are getting a "Restricted Access Page" in local development. This is happening because our test URL for Playwright tests is (0.0.0.0:59836).

I'm looking for advice on how others manage e2e tests with Form Builder under similar constraints. We use the Production Builder License - Monthly Subscription since we are in our project's development/demo phase. Any shared experiences or tips on overcoming these license validation challenges for comprehensive testing in a way that respects the licensing terms would be greatly appreciated:

- What strategies or workarounds have worked for you for successful e2e testing with Form Builder?
- Advice for configuring tests to bypass license restrictions locally and in CI/CD?

**Comment by adamberecz** [comment, marked as answer]
Hi Victor, I've added `0.0.0.0` to our whitelist. If you have a Production Builder License it should work now. Please let me know if you experience any issues.

**Reply by victorvaldez5** [reply]
Thanks, @adamberecz. I appreciate the quick turnaround! Would you also be able to add `127.0.0.1` to the whitelist too?

**Reply by victorvaldez5** [reply]
I just tried running Playwright on the page again, and it still gave me the same error. Is there anything I need to do on the builder config to make it work? Also one thing to note is that the port changes when I run the playwright tests

**Reply by adamberecz** [reply]
127.0.0.1 should already be a part of our whitelist. Are you sure you've followed through the steps for the licensed version (eg. setting up the right registry in `.npmrc`? The steps can be found after you [sign in](https://app.vueform.com/) to you account, then click on Installation at your project then choosing Licensed tab from the left side. The ports should not be an issue.

**Reply by victorvaldez5** [reply]
I can confirm that it was not working because of an error on my end. I am no longer getting a restricted access error. Thanks!
